Pynq-Z1/sdk/pmod_tc1/src/pmod_tc1.c

```diff
@@ -0,0 +1,130 @@
+ *
+ *  1.  Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *
+ *  2.  Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in the
+ *      documentation and/or other materials provided with the distribution.
+ *
+ *  3.  Neither the name of the copyright holder nor the names of its
+ *      contributors may be used to endorse or promote products derived from
+ *      this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ *  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ *  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ *  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ *  OR BUSINESS INTERRUPTION). HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ *  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *****************************************************************************/
+/******************************************************************************
+ *
+ *
+ * @file pmod_tc1.c
+ *
+ * IOP code (MicroBlaze) for Pmod TC1
+ * Pmod TC1 is read only, and has SPI with MISO only (no MOSI)
+ * Operations implemented in this application:
+ *   1. Simple, single read from sensor, and write to data area
+ *   2. Continuous read from sensor and log to data area
+ * Switch configuration is done within this program, Pmod should
+ * be plugged into upper row of connector.
+ * The Pmod TC1 is based on MAX31855 thermocouple-to-digital converter.
+ * http://store.digilentinc.com/pmodtc1-k-type-thermocouple-module-with-wire/
+ *
+ * <pre>
+ * MODIFICATION HISTORY:
+ *
+ * Ver   Who   Date     Changes
+ * ----- ---   ------- -----------------------------------------------
+ * 1.00  tfors 10/15/16 release
+ *
+ * </pre>
+ *
+ *****************************************************************************/
+
+#include "pmod.h"
+
+// MAILBOX_WRITE_CMD
+#define READ_SINGLE_VALUE 0x3
+#define READ_AND_LOG      0x7
+// Log constants
+#define LOG_BASE_ADDRESS (MAILBOX_DATA_PTR(4))
+#define LOG_ITEM_SIZE sizeof(u32)
+#define LOG_CAPACITY  (4000/LOG_ITEM_SIZE)
+
+u32 get_sample(){
+  /*
+   * TC1 data is packed in a 32-bit word.
+   * Four bytes need to be read.
+   */
+  u8 raw_data[4];
+  spi_transfer(SPI_BASEADDR, 4, raw_data, NULL);
+  u32 v = ( (raw_data[0] << 24) + (raw_data[1] << 16) + (raw_data[2] << 8) + raw_data[3] );
```

Comment:
Hi @tfors 

Line 75 seems to exceed the 79 characters. Can you please adjust it? 

We will probably upload some guidelines later.

Thanks,

Rock